### PR TITLE
 First pass at workbox configuration

### PR DESCRIPTION
**Pull request by devnook**
+ Creates precache manifest in webpack
+ Adds a service worker
+ Uses some (arbitrary for now) strategies for handling requests - the goal is to have a basis for further experimentation
+ simplifies build a bit
+ Adds app-wide init (to initialize sw and router)

### CODE REVIEW

File: src/sw.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// POI: we cache the same content twice - in full urls and in partials.
+workbox.routing.registerRoute(
+  new RegExp('\/([A-Za-z\+]+)$'),
+  workbox.strategies.networkFirst()
+);
+
+// Cache at runtime: images from Cloudinary.
+workbox.routing.registerRoute(
+  new RegExp('https\:\/\/res\.cloudinary\.com\/pieshop\/.*'),
+  workbox.strategies.networkFirst(),
```
Comment (samdutton) [inline src/sw.js]: Redundant comma.

File: webpack.common.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 const path = require('path');
+const { InjectManifest } = require('workbox-webpack-plugin');
```
Comment (samdutton) [inline webpack.common.js]: Linter will complain about space around braces.

**Comment by samdutton** [comment]
Excellent! A couple of nits but LGTM.

File: webpack.common.js (hunk 2 above)
Comment (rowan-m) [reply webpack.common.js]: Will see about getting that better integrated. This feedback should be automated.

**Pull request merged**